addon/validators/presence.js

```diff
@@ -1,12 +1,20 @@
 import buildMessage from 'ember-changeset-validations/utils/validation-errors';
 import { validate } from 'ember-validators';
 
-export default function validatePresence(options) {
+export default function validatePresence(options = {}) {
```

Comment:
@jhitchins88 I think `options = {}` may communicate the wrong shape of incoming data here.  I would prefer to leave it without any defaults.  What do you think?  I think in the else case, we can just check if an object.  I think in the past I did...

```js
} else {
  if (Object.prototype.toString.call(options) === '[object Object]' && options.on === 'string') {
    targets = [options.on];
    delete options.on;
  }
}
```

Reply:
Hmm, I see your point.  I don't think it's terribly confusing, but handling the parameter normalization all in the first if statement is useful for readability.

I think we only need to do a simple check for this. We only need options to not be `null/undefined` to not throw an error, since we're not calling the `options.on` function.  The `validate` will throw an error if `presence` isn't defined in the options anyway, so if it's not a `boolean` or an object with `presence` defined, we should be okay.  This is my thoughts:

```
  let targets;
  if (typeof options === 'boolean') {
    options = { presence: options };
  } else if (options && options.on !== undefined) {
    if (typeof options.on === 'string') {
      targets = [ options.on ]
    } else if (Array.isArray(options.on)) {
      targets = options.on
    }

    delete options.on;
  }
```

This is a little bit better to make sure we don't get a bad targets value (e.g., `validatePresence({ on: null })`) and makes sure we don't send the `on` option to the `validate` function (which shouldn't do anything, but the principle).

Reply:
Yep you are right.  Looks good (w/ the semicolons that is :))